### DISCUSSION TITLE
Does this work on Linux as well

### DISCUSSION THREAD

**Discussion by bhavishyagoyal12** [Petersoj/IQFeed4j#2] [Q&A] [answered]
I tried to run this in windows machine it prefectly works but when i tried to run same on Linux Machine i didn't get Level1 Updates ? I am running in Linux using cross over

**Comment by Petersoj** [comment, marked as answer]
@bhavishyagoyal12 it does indeed run on a Linux machine. It runs flawlessly under Wine (a Windows Emulator). I have a trading algorithm that subscribes to Level 1 data for 1800 tickers and haven't noticed any issues (aside from some bugs that IQFeed is aware of and working to fix right now). I haven't tested the `IQconnect.exe` program in Crossover, but I imagine that as long as Crossover supports network interfacing with the host machine, then it should work fine. Do you have any code snippets you could share? Were you running all this during market hours? Note that if you're listening to `SummaryUpdateField.LAST`, then you only receive qualified trades which only happen during market hours and only for round lot trades.

**Reply by bhavishyagoyal12** [reply]
Thanks for your reply . I don't see any disconnection message but no data comes . Is there any configuration needed to do in crossover ?

`public class TestIQFeed {


	public static void main(String args[]) {
		try {
			IQFeed4j iqFeed4j = new IQFeed4j();
			iqFeed4j.startLevel1Feed();
			
			
			iqFeed4j.level1().selectUpdateFieldNames(SummaryUpdateField.MOST_RECENT_TRADE,
					SummaryUpdateField.MOST_RECENT_TRADE_SIZE, SummaryUpdateField.MOST_RECENT_TRADE_MARKET_CENTER,
					SummaryUpdateField.MOST_RECENT_TRADE_DATE, SummaryUpdateField.MOST_RECENT_TRADE_TIME);

			iqFeed4j.level1().requestWatch("AAPL",
					(fundamentalData) -> System.out.printf("@NQ# most recent %.2f split was on %s.\n"),
					(summaryUpdate) -> System.out.printf("NQ trade occurred with %d shares at $%.4f with "));
			

		} catch (IOException exception) {
			exception.printStackTrace();
		} catch (InterruptedException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}

	}

}`

**Reply by bhavishyagoyal12** [reply]
summaryUpdateCSVMapper in is coming null 
```

private void handleSummaryUpdateMessage(String[] csv, Level1MessageType messageType) {
        try {
            SummaryUpdate summaryUpdate = summaryUpdateCSVMapper.map(csv, 1);
            FeedMessageListener<SummaryUpdate> listener =
                    summaryUpdateListenersOfSymbols.get(summaryUpdate.getSymbol());
            if (listener == null) {
                LOGGER.error("Received SummaryUpdate, but no listener for symbol {} exists!",
                        summaryUpdate.getSymbol());
            } else {
                switch (messageType) {
                    case SUMMARY_MESSAGE:
                        summaryUpdate.setType(SummaryUpdate.Type.SUMMARY);
                        break;
                    case UPDATE_MESSAGE:
                        summaryUpdate.setType(SummaryUpdate.Type.UPDATE);
                        break;
                }

                listener.onMessageReceived(summaryUpdate);
            }
        } catch (Exception exception) {
            LOGGER.error("Could not handle SummaryUpdate message!", exception);
        }
    }
```

and exception is  NullPointer
![image](https://user-images.githubusercontent.com/26223513/136999139-7852895e-88e7-4654-99cc-92dfdfc72af4.png)

**Reply by bhavishyagoyal12** [reply]
Same things works in windows

**Reply by Petersoj** [reply]
First, your `System.out.printf` lines don't have any object references in the varargs parameters so it's just printing out the string verbatim.
It should look like this:
```java
iqFeed4j.level1().requestWatch("AAPL",
				(fundamentalData) -> System.out.printf("@NQ# most recent %.2f split was on %s.\n", fundamentalData.getSplitFactor1(), fundamentalData.getSplitFactor1Date()),
				(summaryUpdate) -> System.out.printf("NQ trade occurred with %d shares at $%.4f with ", summaryUpdate.getMostRecentTradeSize(), summaryUpdate.getMostRecentTrade()));
```
And what JDK version are you using? Additionally, can you post the full stack trace of the NPE (NullPointerException)? And are you using the latest `IQConnect.exe` client?

**Reply by bhavishyagoyal12** [reply]
sure let me get that

**Reply by bhavishyagoyal12** [reply]
I am so much thankfull of you. it was version issue of IQfeed

**Reply by Petersoj** [reply]
Hey no problem! Glad your issue was resolved. Let me know if you need help with anything else 🙂

**Reply by bhavishyagoyal12** [reply]
sure . Once again thanks a ton  :)
